Starting point for technical instructions (#22)
digital-earths-global-hackathon/hamburg-node: content/howtotech.md

---
title: First steps - How to hack
---

> To make the most of your time analyzing data and minimize technical hiccups along the way, it's essential to start off on the right foot. This guide lays out the key first steps and to-dos to help you set up everything properly and get started quickly.

Reviewer: I don't like this framing. Scientists avoiding technical knowledge is the main reason why their analysis takes so long 😉 

But I see that this wording likely resonates with the audience

Author: I am rephrasing it slightly... 

```suggestion
To make the most of your time analyzing data and minimize technical hiccups along the way, it's key to begin with a solid setup. This guide lays out first steps and to-dos to help you set up everything properly and get started quickly.
```

You need to [register](https://docs.dkrz.de/doc/getting_started/getting-a-user-account/dkrz-user-account.html#creating-a-new-account-from-scratch) here. Mention, that you are part of the global hackathon and [join an existing project](https://docs.dkrz.de/doc/getting_started/getting-a-user-account/dkrz-user-account.html#join-existing-project). The ID for the project is *bb1153*.

## 2. Log in

Once you are registered, use either https://jupyterhub.dkrz.de or `ssh -Y <userid>@levante.dkrz.de` for *shell logins*.

Levante uses the queuing system [SLURM](https://docs.dkrz.de/doc/levante/running-jobs/index.html) for submission, scheduling, execution, and monitoring of jobs. Choose the **interactive** partition for interactive data analysis.

Information about the file system or **where to store what on Levante**
* /home/ for scripts. Backuped, max 30 GB/user
* /work/ for data you need more than 2 weeks (model output/…), project quota, NO BACKUP
* /scratch/ for temporary files, 15 TB quota - DELETION AFTER TWO WEEKS, NO BACKUP

## 3. Set up your python environment

> [*Input needed...*]

Reviewer: I would either tell users how to register their own Jupyter kernel, or point the the official hackathon kernel. Maybe @florianziemen knows if DKRZ will add it to the public list

Author: Maybe start with setting up their own kernel since it may take a while that the official hk25 kernel is set up? 
Once the official one is set, we can easily update.

### Simulation

> Simulations are stored in [????]
> 
> We are using the [Intake](https://easy.gems.dkrz.de/Processing/Intake/index.html) catalog to store a diversity of data lists. Intake uses xarray for easy data access in python and allows for powerful searches.
> 

Reviewer: ```suggestion
We are using the [Intake](https://easy.gems.dkrz.de/Processing/Intake/index.html) catalog to store a diversity of data lists. Intake uses xarray for easy data access in python and allows for powerful searches.

**The url to the catalog and usage examples will follow shortly before the hackathon.**
```
